internal/cmd/config/config.go

```diff
@@ -441,6 +431,22 @@ func Parse(d string) (*Config, error) {
 	return result, nil
 }
 
+func parsePath(urlField string) (string, error) {
+	// input is returned in error states to preserve `parseutil.ParsePath` functionality.
+	field, err := parseutil.ParsePath(urlField)
+	if err != nil && !errors.Is(err, parseutil.ErrNotAUrl) {
+		return urlField, err
+	}
+	if field != strings.ToLower(field) {
+		return urlField, errors.New("field must be all lower-case")
+	}
+	if !strutil.Printable(field) {
+		return urlField, errors.New("field contains non-printable characters")
+	}
```

Comment:
I am not sure that these checks belong here. While this DRYs things up, in particular the lower case check may not apply in all cases where we might want to use `parseutil.ParsePath`. Similarly, while it probably makes sense to check for non-printable characters in all cases, it a separate concern from what `parseutil.ParsePath` accomplishes.

Reply:
Yep good point. I'm thinking about changing the function signature like this

`func parsePath(urlField string, opts ...parseFieldOpts) (string, error)`

So you could pass some options in to say what you want to be checked. Something like:

```go
parsed, err := parsePath("env://blablabla", withLowerCaseOnlyCheck, withPrintableCheck)
```

I'm a bit torn on this since it seems so overkill for something easy. WDYT?

Reply:
Yea, I think that is overkill, and I think it will still mix concerns. The `parseutil.ParsePath` is concerned with parsing the initial value to extract the true value that might be contained in either an `env://` or `file://` (and in theory additional "paths" in the future). The other two checks are concerned with validation, not parsing. I could potentially see combining those two validation steps into a function, there seems to be a common thing here where a "Name" field should:
- be all lower case
- not contain non-printable characters

But I don't think it should be grouped with the parsing.

Reply:
Perhaps then having another function which enforces these checks that then calls to `parsePath`?

Reply:
Something like this

```go
func parseNamePath(nameField string) (string, error) {
	field, err := parsePath(nameField)
	if err != nil {
		return nameField, err
	}

	if field != strings.ToLower(field) {
		return nameField, errors.New("field must be all lower-case")
	}
	if !strutil.Printable(field) {
		return nameField, errors.New("field contains non-printable characters")
	}

	return field, nil
}

func parsePath(urlField string) (string, error) {
	// input is returned in error states to preserve `parseutil.ParsePath` functionality.
	field, err := parseutil.ParsePath(urlField)
	if err != nil && !errors.Is(err, parseutil.ErrNotAUrl) {
		return urlField, err
	}

	return field, nil
}
```